### ISSUE TITLE
Open file larger than 4 mb in online web word

### ISSUE THREAD

**Issue by ishanig11** [OfficeDev/office-js#4697]
RichApi.Error while opening large documents greater than 4 mb.

Can you please tell me how to achieve this?

**Comment by xiruatms** [comment]
@ishanig11 ， thanks for reporting this. Will you please provide more details?
1. Which API throws this Error please?
2. Are there any codes snippets for us to verify?

**Comment by ishanig11** [comment]
Code snippet:

await Word.run(async (context) => {
        context.document.body.clear();
        const insertFileOptions = {
          importCustomProperties: true,
          importDifferentOddEvenPages: true, // Set to true to import styles from the source document
        };
        context.document.insertFileFromBase64(fileContent, Word.InsertLocation.replace, insertFileOptions);
        await context.sync();
}
insertFileFromBase64 API throws error.
This issue is coming on Web.

**Comment by xiruatms** [comment]
@ishanig11 , thanks for the details.

What makes you think it's the size which caused this error? Any document which is larger than 4mb will cause this error? Will you please provide a template file which caused this eror? Which application are you using please? Sharepoint, OneDrive, WOPI?

**Comment by ishanig11** [comment]
yes, I am facing this issue with any document larger than 4 MB. My document contains images also.

Sorry, I can't share templates as thee files are confidential.

**Comment by shanshanzheng-dev** [comment]
Hi @ishanig11 could you share us error message? It works fine on Word desktop? Could you help to confirm if your document size not large 4MB, it'll not hit that error, right? Thanks.

**Comment by ishanig11** [comment]
Its working fine on Word desktop. Not working on online version. Office throws RichAPI error.

**Comment by jaiswalshubh1** [comment]
hi @ishanig11 ,

Thank you for your patience. As per the internal ticket: 9184071 and after reviewing the reported behavior, we have confirmed that the product is currently functioning as designed. Since this would require a change to the existing product behavior, we are treating it as a feature request rather than a product bug. I am closing the issue.
If you would like us to consider this enhancement for a future release, please submit your suggestion to the Microsoft Tech Community Ideas Forum or upvote an existing idea. This allows other developers with similar requirements to provide feedback and helps our product teams assess community interest.🔗 Ideas Forum:
[https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/idb-p/Microsoft365Developer…](https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/idb-p/Microsoft365Developer%E2%80%A6)

Thank you for your understanding and support!

CC: @kustonaut 

Thanks!
Shubham Jaiswal
